base/intfuncs.jl

```diff
@@ -325,7 +325,8 @@ end
 
 function digits!{T<:Integer}(a::AbstractArray{T,1}, n::Integer, base::Integer=10)
     2 <= base || throw(ArgumentError("base must be ≥ 2, got $base"))
-    base - 1 <= typemax(T) || throw(ArgumentError("type $T too small for base $base"))
+    !applicable(typemax, T) ||
```

Comment:
I'm not sure we really want to call `applicable` here: based on the `@code_llvm` it doesn't seem to be evaluated only at compile time. Unless anyone can think of something better, it might be better to just have `T == BigInt || base-1 <= typemax(T) || ... `